cumulus/parachains/runtimes/assets/asset-hub-rococo/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Asset Hub Rococo Runtime genesis config presets
+
+#[cfg(not(feature = "std"))]
+use crate::vec;
+use crate::Vec;
+use cumulus_primitives_core::ParaId;
+use hex_literal::hex;
+use parachains_common::{genesis_config_helpers::*, AccountId, AuraId, Balance as AssetHubBalance};
+use sp_core::{crypto::UncheckedInto, sr25519};
+use testnet_parachains_constants::rococo::xcm_version::SAFE_XCM_VERSION;
+
+const ASSET_HUB_ROCOCO_ED: AssetHubBalance = crate::ExistentialDeposit::get();
+
+/// Generate the session keys from individual elements.
+///
+/// The input must be a tuple of individual keys (a single arg for now since we have just one key).
+pub fn asset_hub_rococo_session_keys(keys: AuraId) -> crate::SessionKeys {
+	crate::SessionKeys { aura: keys }
+}
+
+fn asset_hub_rococo_genesis(
+	invulnerables: Vec<(AccountId, AuraId)>,
+	endowed_accounts: Vec<AccountId>,
+	endowment: AssetHubBalance,
+	id: ParaId,
+) -> serde_json::Value {
+	serde_json::json!({
+		"balances": crate::BalancesConfig {
```

Comment:
Why are we not using the concrete `GenesisConfig` type here? And then transform this into a json?

Reply:
A preset is meant to be patch. This is the only reason.

If `GenesisConfig` struct would be used here, then the preset would contain all the fields of the genesis config. That would be less convenient / a little harder to read  when displaying presets with some external tools (e.g. `chain-spec-builder`).